src/activities/ActivityUsageEntity.js

```diff
@@ -687,10 +687,9 @@ export class ActivityUsageEntity extends Entity {
 			[this.inGrades(), activity.scoreAndGrade.inGrades]
 		];
 
+		// For activities that don't use scoreOutOf, this.scoreOutOf() is initialized to undefined
 		const scoreOutOf = `${this.scoreOutOf()}`;
-
-		// For activities that don't use scoreOutOf, this.scoreOutOf() is initialized to undefined whereas for those that do it initializes to ''
-		if (scoreOutOf !== 'undefined') {
+		if (scoreOutOf !== undefined) {
```

Comment:
Is this check correct? Will `scoreOutOf` ever be `undefined` here? Regardless of how it's initialized it will be parsed as a string on 691 right? 

I'm thinking of the case when `this.scoreOutOf()` returns `undefined`. We'll be comparing the string `'undefined'` with the value `undefined` and end up with the string added to the diffs. Is that intended?

Reply:
When `this.scoreOutOf()` is called, doesn't it possibly return `undefined` and not the string `"undefined"`?
That's why I changed this to check for the non-string version.

Reply:
Yeah but even if the function call returns the value `undefined` we are then parsing it as a string by using the template literal syntax on line 691

![image](https://user-images.githubusercontent.com/46040098/119731474-d74dbf80-be2b-11eb-93ed-30fdd54af8ce.png)

Reply:
imo intent would be clearer doing something like this:

```js
const scoreOutOf = this.scoreOutOf();
if (scoreOutOf !== undefined) {
  diffs.push([String(scoreOutOf), activity.scoreAndGrade.scoreOutOf]);
}
```
so we actually check the value returned from the function and not the stringified version and then explicitly cast it to a string when adding to diffs 🤷

Reply:
yes